tests/utils/test_operation.py

```diff
@@ -94,3 +120,69 @@ def test_monitor():
     succeeded_status = OperationState[op_succeeded.state]
     assert op.resource_id == op_succeeded.resource_id
     assert succeeded_status == OperationState.SUCCEEDED
+
+
+@mock_api()
+def test_wait():
+    client = utils.client.create(**CONFIG["my_instance_name"])
+    project = client.projects.by_resource_id(CONFIG["projects"]["minimal_mastering"])
+    project = project.as_mastering()
+    op = project.pairs().refresh(asynchronous=True)
+    op_after_wait = wait(op, poll_interval_seconds=1)
+    succeeded_status = OperationState[op_after_wait.state]
+    assert op.resource_id == op_after_wait.resource_id
+    assert succeeded_status == OperationState.SUCCEEDED
+
+
+@mock_api()
+def test_get_safe_pair_estimate():
+    client = utils.client.create(**CONFIG["my_instance_name"])
+
+    mastering_project = client.projects.by_resource_id(
+        CONFIG["projects"]["minimal_mastering"]
+    )
+    estimate_counts_op = safe_estimate_counts(mastering_project)
+
+    assert OperationState[estimate_counts_op.state] == OperationState.SUCCEEDED
+
+
+@mock_api()
+def test_timeout_error():
+    def mock_timeout_fcn(_op: tamr_unify_client.operation.Operation) -> None:
+        sleep(2)
+        return _op
+
+    client = utils.client.create(**CONFIG["my_instance_name"])
+    project = client.projects.by_resource_id(CONFIG["projects"]["minimal_mastering"])
+    project = project.as_mastering()
+    op = project.pairs().refresh(asynchronous=True)
+    with pytest.raises(TimeoutError, match="Waiting for operation took longer than"):
+        with mock.patch.object(
+            tamr_unify_client.operation.Operation,
+            attribute="poll",
+            new=mock_timeout_fcn,
+        ):
+            monitor(op, timeout_seconds=1)
+
+
+def test_none_status():
+    client = utils.client.create(**CONFIG["my_instance_name"])
+    op = from_resource_id(client, job_id="-1")
+
+    with mock.patch.object(
+        tamr_unify_client.operation.Operation, attribute="status", new=None
```

Comment:
This is a bit convoluted in an attempt to test the `operation.status is None` cases in the monitor/wait functions. I don't know how that would arise in practice so I put some mocks in to generate. If anyone knows how such an op can arise, please let me know. It would be better to test a realistic scenario